django/http/request.py

```diff
@@ -362,32 +358,26 @@ def _load_post_and_files(self):
             self._mark_post_parse_error()
             return
 
-        if self.content_type == "multipart/form-data":
-            if hasattr(self, "_body"):
-                # Use already read data
-                data = BytesIO(self._body)
-            else:
-                data = self
+        # TODO create a parsers setter/getter/initializer like upload_handlers
+        parser_list = (parsers.FormParser(), parsers.MultiPartParser())
+
+        selected_parser = None
+        for parser in parser_list:
+            if self.content_type == parser.media_type:
+                selected_parser = parser
+                break
```

Comment:
Just checking… This should definitely be a method on `parser` so that arbitrary logic can be brought to bear.

Reply:
Yes. The penny dropped for me this morning on this following the discussion on the `can_handle(...)` method.

Reply:
In [0fd67b8](https://github.com/django/django/pull/17546/commits/0fd67b89efbf17fef811dc18652b7d9926d05282) I updated this to:
```python
            if parser.can_handle(self.content_type):
                selected_parser = parser
                break
```